sdk/core/System.ClientModel/src/Options/RequestOptions.cs

```diff
@@ -19,14 +20,12 @@ public class RequestOptions
     private PipelinePolicy[]? _perTryPolicies;
     private PipelinePolicy[]? _beforeTransportPolicies;
 
-    private readonly PipelineRequestHeaders _addHeaders;
+    private List<HeadersUpdate>? _headersUpdates;
```

Comment:
Discussed offline with @KrzysztofCwalina whether he's concerned about this collection growing in size -- we decided that in practice it shouldn't be an issue, so this should be fine.